### PR TITLE
REPL: make terminating with a trailing colon display output with 1000 row limit

### PR DESCRIPTION
**Pull request by IanButterworth**
Updated to a single trailing `:`
____

This is a proposal to make terminating input in the repl with `;;` a convenient way to `display` output without row limit.

Sometimes you just want to see the entire object, printed in the nice formatting that `display` provides, without truncating to the terminal window height.

When generating this the terminal window was quite short to illustrate the effect.

```
julia> d = Dict(i => i for i in 1:20)
Dict{Int64, Int64} with 20 entries:
  5  => 5
  16 => 16
  7  => 7
  20 => 20
  12 => 12
  8  => 8
  17 => 17
  1  => 1
  ⋮  => ⋮

julia> 
```
A single `;` still suppresses output.
```
julia> d;

julia> 
```
Double `;;` allows `display` to show all rows (requiring the user to scroll up)
```
julia> d;;
Dict{Int64, Int64} with 20 entries:
  5 => 5
  16 => 16
  7 => 7
  20 => 20
  12 => 12
  8 => 8
  17 => 17
  1 => 1
  19 => 19
  4 => 4
  6 => 6
  13 => 13
  2 => 2
  10 => 10
  11 => 11
  9 => 9
  15 => 15
  18 => 18
  14 => 14
  3 => 3

julia>
```

### CODE REVIEW

**Comment by JeffBezanson** [comment]
This is a good feature but triage does not like using more semicolons to trigger it.

**Comment by Seelengrab** [comment]
This could be done as a REPL setting ala #41435, e.g. to enable the pager.

**Comment by jariji** [comment]
I have `showall(x) = show(stdout, "text/plain", x)`  in my startup.jl for this.

**Comment by IanButterworth** [comment]
I'll try to not die on the hill of `;;`, but my proposal is to make `;` in the repl conceptually an _output modifier_. 0 is default, 1 suppresses, 2 makes verbose, 3 ...

Did triage happen to think of any other trailing etc. chars that could be used?

**Comment by IanButterworth** [comment]
I've switched it over to if there's a trailing space. 

Also, the limit now extends to 1000 rows, for safety if a very large object is displayed accidentally.

**Comment by jishnub** [comment]
I find this behavior somewhat unintuitive, as I'd expect invisible whitespace characters to not change the output. Can't this setting be more explicit, perhaps something like `np.set_printoptions`?

**Comment by IanButterworth** [comment]
Putting this setting under a menu does seem reasonable, I just prefer the ease of doing it very easily, something like this PR does. 

If there was an options repl added, it could definitely go in there as a sticky setting too. I just think it should be achievable with some simple instruction on an individual input.

**Comment by IanButterworth** [comment]
How do people feel about a trailing colon?

```
julia> x
{default show limited}

julia> x;
{no show}

julia> x:
{show all (1000 row limit)}
```

**Comment by IanButterworth** [comment]
Marking this for a hopefully brief re-triage for the idea of using a trailing colon.

File: stdlib/REPL/src/REPL.jl [hunk 1]
```diff
@@ -277,11 +277,25 @@ function display(d::REPLDisplay, mime::MIME"text/plain", x)
...
     repl.waserror = response[2]
     with_repl_linfo(repl) do io
         io = IOContext(io, :module => active_module(repl)::Module)
-        print_response(io, response, show_value, have_color, specialdisplay(repl))
+        if limitrows
+            print_response(io, response, show_value, have_color, specialdisplay(repl))
+        else
+            before = get(Base.active_repl.options.iocontext, :displaysize, nothing)
+            try
+                Base.active_repl.options.iocontext[:displaysize] = (1000, displaysize(io)[2])
```
Comment (StefanKarpinski) [inline stdlib/REPL/src/REPL.jl:289]: Should the 1000 value be configurable somehow? Someone will inevitably ask...

**Comment by oscardssmith** [comment]
triage still doesn't like this with trailing `:` or the fixed number limit and thinks this should be a macro of some sort.
Possibly, this should be `@pretty x` (similar to `@show` but with pretty-printing). The name is highly bikesheddable.

**Pull request closed without merging**